pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -1308,20 +1306,38 @@ func (cloud *CloudProvider) CreateSnapshot(ctx context.Context, project string,
 		return nil, err
 	}
 
+	snapshotToCreate := &computev1.Snapshot{
+		Name:             snapshotName,
+		StorageLocations: snapshotParams.StorageLocations,
+		Description:      description,
+		Labels:           snapshotParams.Labels,
+		SourceDisk:       cloud.GetDiskSourceURI(project, volKey),
+	}
 	switch volKey.Type() {
 	case meta.Zonal:
 		if description == "" {
 			description = "Snapshot created by GCE-PD CSI Driver"
 		}
-		return cloud.createZonalDiskSnapshot(ctx, project, volKey, snapshotName, snapshotParams, description)
 	case meta.Regional:
 		if description == "" {
 			description = "Regional Snapshot created by GCE-PD CSI Driver"
```

Comment:
It doesn't look like you're setting description on the request. Also I'd invert the description empty string check, and the volKey Type check.

Reply:
what do you mean by
> Also I'd invert the description empty string check, and the volKey Type check.

Thanks for the description catch.